linera-explorer/src/lib.rs

```diff
@@ -328,65 +331,55 @@ fn set_onpopstate(app: JsValue) {
     callback.forget()
 }
 
-/// Initialize pages and subscribe to notifications
+/// Initializes pages and subscribes to notifications
 #[wasm_bindgen]
 pub async fn init(app: JsValue, uri: String) {
     console_error_panic_hook::set_once();
     set_onpopstate(app.clone());
     let data = from_value::<Data>(app.clone()).expect("cannot parse vue data");
-    let address = node_service_address(&data.config, false);
+    let address = node_service_address(&data.config, Protocol::Http);
     let default_chain = chains(&app, &address).await;
     match default_chain {
         Err(e) => {
             route_aux(
                 &app,
                 &data,
                 &Some("error".to_string()),
-                &data.chain,
                 &[("msg".to_string(), e.to_string())],
             )
             .await;
         }
         Ok(default_chain) => {
             let uri = Url::parse(&uri).expect("failed to parse url");
             let pathname = uri.path();
-            let mut queries = uri.query_pairs();
-            let chain_id = queries
-                .find_map(|(k, v)| {
-                    if k == "chain" {
-                        Some(ChainId::from_str(&v).expect("cannot build chainId"))
-                    } else {
-                        None
-                    }
-                })
-                .unwrap_or(default_chain);
-            let (path, args) = match pathname {
-                "/blocks" => (Some("blocks".to_string()), vec![]),
-                "/applications" => (Some("applications".to_string()), vec![]),
+            let mut args : Vec<(String, String)> = uri.query_pairs().into_owned().collect();
+            args.push(("chain".to_string(), default_chain.to_string()));
```

Comment:
(If there's already a `chain` argument, the existing one will take precedence, right? Because we'll find it first and use it.)